jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -268,7 +268,134 @@ private static StringBuilder encodePath(StringBuilder buf, String path, int offs
 
         return buf;
     }
-    
+
+    /**
+     * Encode a raw URI String and convert any raw spaces to
+     * their "%20" equivalent.
+     *
+     * @param str input raw string
+     * @return output with spaces converted to "%20"
+     */
+    public static String encodeSpaces(String str)
+    {
+        return StringUtil.replace(str, " ", "%20");
+    }
+
+    /**
+     * Encode a raw String and convert any specific characters to their URI encoded equivalent.
+     *
+     * @param str input raw string
+     * @param charsToEncode the list of raw characters that need to be encoded (if encountered)
```

Comment:
I think the chars in this string should be restricted to ones that must be encoded in URLs anyway.  That way, most a-zA-Z0-9 characters will not need to check the charsToEncode string

Reply:
Unfortunately, things like the Mongo utils wants specific "special" characters encoded, ones that are not normally URI encoded.

For example: Mongo wants "." (dot) and "%" (percent) encoded.
And WebAppClassloader wants "," (comma) and ";" (semicolon) encoded.